### PR TITLE
*: support getting role in auth v3

### PR DESCRIPTION
**Pull request by mitake**
This commit implements RoleGet() RPC of etcdserver and adds a new
subcommand "role get" to etcdctl v3. It will list up permissions that
are granted to a given role.

$ ETCDCTL_API=3 bin/etcdctl role get r1
Role r1
KV Read:
        b
        d
KV Write:
        a
        c
        d

### CODE REVIEW

**Comment by mitake** [comment]
@xiang90 the test failure seems to be nondeterministic. On my machine, `INTEGRATION=y ./test` passed successfully.

File: etcdctl/ctlv3/command/role_command.go [hunk 1]
```diff
@@ -83,3 +92,29 @@ func roleGrantCommandFunc(cmd *cobra.Command, args []string) {
...
+		ExitWithError(ExitBadArgs, fmt.Errorf("role get command requires role name as its argument."))
+	}
+
+	resp, err := mustClientFromCmd(cmd).Auth.RoleGet(context.TODO(), args[0])
+	if err != nil {
+		ExitWithError(ExitError, err)
+	}
+
+	fmt.Printf("Role %s\n", args[0])
+	fmt.Printf("KV Read:\n")
```
Comment (xiang90) [inline etcdctl/ctlv3/command/role_command.go]: println?

File: etcdctl/ctlv3/command/role_command.go [hunk 2]
```diff
@@ -83,3 +92,29 @@ func roleGrantCommandFunc(cmd *cobra.Command, args []string) {
...
+	for _, perm := range resp.Perm {
+		if perm.PermType == clientv3.PermRead || perm.PermType == clientv3.PermReadWrite {
+			fmt.Printf("\t%s\n", string(perm.Key))
+		}
+	}
+	fmt.Printf("KV Write:\n")
```
Comment (xiang90) [inline etcdctl/ctlv3/command/role_command.go]: println?

**Comment by xiang90** [comment]
lgtm

**Comment by mitake** [comment]
@xiang90 updated for the println, PTAL.

**Comment by xiang90** [comment]
will merge when CI turns green. there are a few known flaky tests. i will re-trigger CI if needed.

**Pull request merged**